app/controllers/CheckingServiceController.scala

```diff
@@ -48,66 +49,60 @@ trait CheckingServiceController extends ERSCheckingBaseController {
 
   def showStartPage()(implicit request: Request[AnyRef], messages: Messages): Future[Result] = Future.successful(Ok(views.html.start(request, context, messages)))
 
-  def schemeTypePage(): Action[AnyContent] = authAction.async {
+  def schemeTypePage(form: Form[CS_schemeType] = CSformMappings.schemeTypeForm): Action[AnyContent] = authAction.async {
       implicit request =>
-        showSchemeTypePage(request, hc)
+        showSchemeTypePage(form: Form[CS_schemeType])(request, hc)
   }
 
-  def showSchemeTypePage(implicit request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = {
-    val schemeType: String = "0"
-    Future(Ok(views.html.scheme_type(schemeType)))
+  def showSchemeTypePage(form: Form[CS_schemeType])(implicit request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = {
+    Future(Ok(views.html.scheme_type(form)))
   }
 
   def schemeTypeSelected(): Action[AnyContent] = authAction.async {
       implicit request =>
         showSchemeTypeSelected(request)
   }
 
-  def showSchemeTypeSelected(implicit request: Request[AnyRef]): Future[Result] = {
+  def showSchemeTypeSelected(implicit request: Request[AnyContent]): Future[Result] = {
     CSformMappings.schemeTypeForm.bindFromRequest.fold(
       formWithErrors => {
-        Future.successful(Redirect(routes.CheckingServiceController.schemeTypePage).flashing("scheme-not-selected-error" -> Messages("ers_scheme_type.select_scheme_type")))
+        schemeTypePage(formWithErrors).apply(request)
```

Comment:
Hmm by using this function on an error scenario that means that we are treating an error as an OK response which doesn't seem right to me. I don't remember whether it's usually a Redirect or BadRequest but it's not usually OK.

Reply:
on it - my bad

Reply:
updated to give a bad request and send user straight to the view